js/gimmicks/chart.js

```diff
@@ -0,0 +1,111 @@
+                chartOptions: {
+                    responsive: true
+                }
+            };
+            var options = $.extend ({}, default_options, opt);
+
+            // Replace the Gimmick with the canvas that chartJS needs
+            var $link = $(link);
+            var myHtml = $('<canvas id="' + options.canvasId + '" ></canvas>');
+            $link.replaceWith(myHtml);
+
+            // This is the object that is given to the chart frame work for rendering. It will be
+            // built up by processing the html table that is found on the table.
+            var chartConfig = {
+                "datasets": [],
+                "labels": []
+            };
+
+            // These variables hold the indices of the columns in the table we care about. They will
+            // be populated as we process the table based on the options that are given in the
+            // gimmick
+            var labelColumnIndex = -1;
+            var dataColumnIndices = [];
+
+            // Get the index of the columns that we care about for charting
+            // TODO JC Currently assumes that there is a single table on the page. Document this or
+            //         implement a way to indicate which table
+            jQuery("th").each(function(index){
+
+                // This is the column that labels each data point
+                if(this.textContent === options.labelColumn) {
+                    labelColumnIndex = index;
+                }
+
+                // Check if this is a data column
+                else {
+                    for(var i = 0; i < options.dataColumns.length; i++) {
+                        if(this.textContent === options.dataColumns[i]) {
+                            dataColumnIndices.push(index);
+                        }
+                    }
+                }
+            });
+
+            // Get the data
+            jQuery("tr").each(function(rowIndex){
+                $(this).find("td").each(function(colIndex){
+
+                    if(colIndex === labelColumnIndex) {
+                        chartConfig.labels.push(this.textContent);
+                    } else {
+                        for(var i = 0; i < dataColumnIndices.length; i++) {
+                            if(colIndex === dataColumnIndices[i]) {
+
+                                if(chartConfig.datasets[i] === undefined) {
+                                    chartConfig.datasets[i] = {};
+                                    chartConfig.datasets[i].data = [];
+                                }
+
+                                chartConfig.datasets[i].data.push(this.textContent);
+                            }
+                        }
+                    }
+                });
+            });
+
+            setTimeout(function(){
+                new Chart(document.getElementById(options.canvasId).getContext("2d"))[options.chartType](chartConfig, options.chartOptions);
+            }, 100);
```

Comment:
We can get rid of the setTimeout by using the (sadly undocumented) way of MDwiki to subscribe to the "execution phases". We subscribe into the "postgimmick" phase, which will called after the 'gimmick' phase that I mentioned earlier. Example code:

```js
$.md.stages('postgimmick').subscribe(function() {
    new Chart(document.getElementById(options.canvasId).getContext("2d"))[options.chartType](chartConfig, options.chartOptions);
});
```